adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/MessageMapping.java

```diff
@@ -31,15 +31,23 @@
 
     /**
      * Maps a message uploaded by a device.
+     * <p>
+     * If a mapping service is not configured for a gateway or a protocol adapter, this method returns
+     * a successful future containing a {@code null} value. If a mapping service is configured for a gateway
```

Comment:
why do we need to introduce this additional case that the client needs to check for? Can't we simply do the same as for a 2xx != 200 status code, i.e. completing the future with a mapped message containing the original device payload and target address? That way the client wouldn't need to handle a `null` result specifically, would it?

Reply:
make sense. i was interpreting this requirement you stated in the issue

> No mapping service is configured for the gateway -> Nothing to be done at all

I interpreted doing nothing to mean returning null as a way to inform the client that nothing has been done. but will fix